reports/lib-utils.api.md

```diff
@@ -0,0 +1,837 @@
+
+// @public (undocumented)
+export const commonFetchJSON: {
+    <TResult>(url: string, requestInit?: RequestInit | undefined, timeout?: number | undefined, isK8sAPIRequest?: boolean | undefined): Promise<TResult>;
+    put<TResult_1>(url: string, data: unknown, requestInit?: RequestInit | undefined, timeout?: number | undefined, isK8sAPIRequest?: boolean | undefined): Promise<TResult_1>;
+    post<TResult_2>(url: string, data: unknown, requestInit?: RequestInit | undefined, timeout?: number | undefined, isK8sAPIRequest?: boolean | undefined): Promise<TResult_2>;
+    patch<TResult_3>(url: string, data: unknown, requestInit?: RequestInit | undefined, timeout?: number | undefined, isK8sAPIRequest?: boolean | undefined): Promise<TResult_3>;
+    delete<TResult_4>(url: string, data?: unknown, requestInit?: RequestInit | undefined, timeout?: number | undefined, isK8sAPIRequest?: boolean | undefined): Promise<TResult_4>;
+};
+
+// @public (undocumented)
+export const commonFetchText: (url: string, requestInit?: RequestInit | undefined, timeout?: number | undefined, isK8sAPIRequest?: boolean | undefined) => Promise<string>;
+
+// @public
+export type DestroyHandler = GenericHandler<unknown | undefined>;
+
+// @public (undocumented)
+export const DetailsItem: React_3.FC<DetailsItemProps>;
+
+// @public (undocumented)
+export const DetailsItemList: React_3.FC;
+
+// @public (undocumented)
+export type DetailsItemProps = {
+    defaultValue?: string;
+    hideEmpty?: boolean;
+    loaded?: boolean;
+    label: string;
+    resource?: K8sResourceCommon;
+    path?: string | string[];
+};
+
+// @public (undocumented)
+export const DetailsPage: (props: Omit<DetailsPageProps, keyof WithRouterProps>) => JSX.Element;
+
+// @public (undocumented)
+export type DetailsPageHeaderProps = {
+    breadcrumbs: BreadcrumbProp[];
+    actionButtons?: ActionButtonProp[];
+    pageHeading?: PageHeading;
+    obj?: K8sResourceCommon;
+    actionMenu?: ActionMenuProps;
+};
+
+// @public (undocumented)
+export type DetailsPageProps = HorizontalNavProps & DetailsPageHeaderProps;
+
+// @public (undocumented)
+export type DiscoveryResources = {
+    adminResources: string[];
+    allResources: string[];
+    configResources: K8sModelCommon[];
+    clusterOperatorConfigResources: K8sModelCommon[];
+    models: K8sModelCommon[];
+    namespacedSet: Set<string>;
+    safeResources: string[];
+    groupVersionMap: {
+        [key: string]: {
+            versions: string[];
+            preferredVersion: string;
+        };
+    };
+};
+
+// @public (undocumented)
+export type ErrorHandler = GenericHandler<Event>;
+
+// @public (undocumented)
+export type FilterItem = {
+    label: string;
+    id: string;
+};
+
+// @public (undocumented)
+export type FilterValue = Partial<{
+    selected: string[];
+    all: string[];
+}>;
+
+// @public (undocumented)
+export type GenericHandler<T = unknown> = (data: T) => void;
+
+// @public (undocumented)
```

Comment:
@vojtechszocs I'm assuming we need to take todos to document these undocumented types?